lib/useOnyx.ts

```diff
@@ -252,10 +264,10 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(
             return result;
         }
 
-        // We get the value from cache while the first connection to Onyx is being made,
+        // We get the value from cache while the first connection to Onyx is being made or the key has changed,
```

Comment:
Why do we return the cached value if the key has changed? Can we include the reason in this comment?
```suggestion
        // We get the value from cache while the first connection to Onyx is being made or if the key has changed,
```